### PR TITLE
feat: ems plugin design changes

### PR DESCRIPTION
**Pull request by rahulguptajss**

### CODE REVIEW

File: cmd/poller/collector/collector.go [hunk 1]
```diff
@@ -521,6 +532,8 @@ func (me *AbstractCollector) LoadPlugins(params *node.Node, c Collector) error {
 
 	var p plugin.Plugin
 	var abc *plugin.AbstractPlugin
+	var plugins []plugin.Plugin
+	me.Plugins = make(map[string][]plugin.Plugin)
```
Comment (Hardikl) [inline cmd/poller/collector/collector.go:536]: would be better to move this initialize line at 198

Comment (Hardikl) [reply cmd/poller/collector/collector.go:536]: rest look good as per the meeting discussion

File: cmd/poller/collector/collector.go [hunk 2]
```diff
@@ -365,21 +368,29 @@ func (me *AbstractCollector) Start(wg *sync.WaitGroup) {
 			}
 
 			if data != nil {
-				results = append(results, data)
+
+				v := make([]*matrix.Matrix, 0, len(data))
```
Comment (cgrinds) [inline cmd/poller/collector/collector.go:372]: I think these lines can be simplified to
```go
for _, value := range data {
    results = append(results, value)
}
```

**Review by cgrinds — COMMENTED** [review]
looks good @rahulguptajss

**Review by cgrinds — APPROVED** [review]

**Pull request merged**